src/main/java/duke/Duke.java

```diff
@@ -46,7 +60,53 @@ public void run() {
             }
         }
     }
-    public static void main(String[] args) {
-        new Duke("data/duke.txt").run();
+
+    public String getResponse(String fullCommand) {
+        // Create a stream to hold the output
+        ByteArrayOutputStream narutoStream = new ByteArrayOutputStream();
+        PrintStream ps = new PrintStream(narutoStream);
+        // IMPORTANT: Save the old System.out!
+        PrintStream old = System.out;
+        // Tell Java to use your special stream
+        System.setOut(ps);
+
+        // Now execute the entire command from user. It goes to my special stream
+        try {
+            Command c = Parser.parse(fullCommand);
+            c.execute(tasks, ui, storage);
+        } catch (DukeException e) {
+            ui.showError(e);
+        } finally {
+            ui.showLines();
+        }
+
+        // Put things back. Important!
+        System.out.flush();
+        System.setOut(old);
+        // Show what happened in the terminal on IntelliJ
+        System.out.println("Here: " + narutoStream.toString());
+
+        return "Naruto's reply! " + narutoStream.toString();
     }
+
+    public String initialMessageFromNaruto() {
+        // Create a stream to hold the output
+        ByteArrayOutputStream narutoStream = new ByteArrayOutputStream();
+        PrintStream ps = new PrintStream(narutoStream);
+        // IMPORTANT: Save the old System.out!
+        PrintStream old = System.out;
```

Comment:
a better name might be "oldPrintStream".

Reply:
That's true!